org-cyf-guides/content/pair-programming/index.md

```diff
@@ -0,0 +1,37 @@
++++
+title = 'Pair programming'
+description = 'Practice writing code and getting support'
+emoji = '👥'
++++
+
+Pair programming is an excellent way to develop programming and communication skills.
+
+It is often much easier to work through something when working on something 1-to-1.
+
+It is also helps our trainees to prepare for technical interviews when they'll need to code in front of other people.
+
+## What
+
+Pair programming is working together with someone else to code a solution to a problem.
+
+Pick a piece of programming to do, find a buddy, and work on it together.
+
+You can do this in person (sharing one computer), or remotely (either by sharing your screen, or using something like [CodeTogether](https://marketplace.visualstudio.com/items?itemName=genuitecllc.codetogether)).
+
+Sometimes one person does all of the writing, and has discussion with the other. Other times, people take turns writing.
+
+## Guidance for trainees
+
+When approaching a problem, you need to **explain your thought process**, **plan out what to do**, **write the code**, and **check that it works**.
+
+## Guidance for volunteers
+
+**Help people learn**: Most of the exercises the trainees are doing aren't useful in their own right. The point is for the trainee to learn and grow through them. Focus on understanding, and techniques that will help solve the next problem.
+
+**Don't take over!** It's important trainees get used to figuring things out. Provide guidance and assistance but trainees need to struggle to overcome any obstacles with understanding and technical communication.
```

Comment:
```suggestion
**Don't take over!** It's important trainees get used to figuring things out. Provide guidance and assistance but trainees need to struggle to overcome any obstacles with understanding and technical communication. You need to sit with their struggle.
```

Reply:
I went with "It can be uncomfortable to watch someone struggle, but make sure they've considered and tried all of their ideas before you intervene."